### PR TITLE
Autodock vina: spec

### PR DESCRIPTION
**Pull request by zenln**

### CODE REVIEW

**Comment by zenln** [comment]
Any comment on this?

File: var/spack/repos/builtin/packages/autodock-vina/package.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    """AutoDock Vina is an open-source program for doing molecular docking"""
+
+    homepage = "http://vina.scripps.edu/"
+    url = "http://vina.scripps.edu/download/autodock_vina_1_1_2.tgz"
+
+    version('1_1_2', 'b467b71ee77dd155b65b1c5364e4220f')
+
+    depends_on('boost')
+
+    patch('main.patch')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/autodock-vina/package.py]: Out of curiosity are these patches you created yourself? I didn't see them when taking a cursory look at the website.

File: var/spack/repos/builtin/packages/autodock-vina/package.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    def build(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make()
+
+    def install(self, spec, prefix):
+        with working_dir(self.build_directory):
+            mkdirp(prefix.bin)
+            copyfile("vina", join_path(prefix.bin, "vina"))
+            copymode("vina", join_path(prefix.bin, "vina"))
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/autodock-vina/package.py]: Spack provides a module-level command 'install' that will handle this for you, could you swap this with calls to `install`?

**Comment by scheibelp** [comment]
Thanks for the bump. I have a question and a minor request (above)

File: var/spack/repos/builtin/packages/autodock-vina/package.py (hunk 2 above)
Comment (junghans) [reply var/spack/repos/builtin/packages/autodock-vina/package.py]: Yeah, an explanation in a comment where the patches come from and why they are needed would be good!
`main.patch` is really not very descriptive.

File: var/spack/repos/builtin/packages/autodock-vina/package.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    """AutoDock Vina is an open-source program for doing molecular docking"""
+
+    homepage = "http://vina.scripps.edu/"
+    url = "http://vina.scripps.edu/download/autodock_vina_1_1_2.tgz"
+
+    version('1_1_2', 'b467b71ee77dd155b65b1c5364e4220f')
+
+    depends_on('boost')
+
+    # Replacing depecrated function call of boost with current function call
```
Comment (junghans) [inline var/spack/repos/builtin/packages/autodock-vina/package.py]: Does that mean it doesn't work with some older version of boost? i.e. `depends('boost@X.Y:')`

Comment (zenln) [reply var/spack/repos/builtin/packages/autodock-vina/package.py]: Good point, I am not sure if this patch will make it not work for older versions.

**Pull request merged**